doc/mkdocs/docs/api/macros/json_legacy_comparison.md

```diff
@@ -0,0 +1,51 @@
+# JSON_LEGACY_COMPARISON
+
+```cpp
+#define JSON_LEGACY_COMPARISON /* value */
+```
+
+This macro enables the (incorrect) legacy comparison behavior of discarded JSON values.
+Possible values are `1` to enable or `0` to disable (default).
+
+When enabled, comparisons involving at least one discarded JSON value yield results as follows:
+
+| **Operator** | **Result**    |
+|--------------|---------------|
+| `==`         | `#!cpp false` |
+| `!=`         | `#!cpp true`  |
+| `<`          | `#!cpp false` |
+| `<=`         | `#!cpp true`  |
+| `>=`         | `#!cpp true`  |
+| `>`          | `#!cpp false` |
+
+Otherwise, comparisons involving involving at least one discarded JSON value always yield `#!cpp false`.
+
+## Default definition
+
+The default value is `0`.
+
+```cpp
+#define JSON_LEGACY_COMPARISON 0
+```
+
+When the macro is not defined, the library will define it to its default value.
+
+## Notes
+
+!!! warning "Inconsistent behavior in C++20-mode"
+
+    The `<=>` operator will always yield the correct result (`#!cpp false`) but overloads
+    are provided so that any comparison not using the `<=>` operator directly will correctly
+    emulate the legacy behavior.
+    Be advised that standard library and third-party code can still use the 3-way comparison
+    operator directly, resulting in inconsistent and unpredictable behavior.
+
+!!! warning "Deprecation"
```

Comment:
I am not 100% happy with the term "deprecation" here. In fact, 3.11.0 will be a breaking change, and this macro is the only way to get back the legacy behavior. I don't find better words though...

Reply:
The idea would be that the pre-3.11.0 behavior, i.e. legacy behavior, can be restored until it is removed in 4.0.0. Hence the macro is deprecated?

Reply:
I have rearranged and reworded the section.

Reply:
It's better, thanks!

The more I think of it, maybe (not in this PR, but for 3.11.0) we should add a dedicated page for comparison of JSON values where, on one page, the core ideas are described. This would also be a place to address https://github.com/nlohmann/json/issues/3443.

Reply:
A folder and category to place general help topics in would also be useful. I've repeatedly discarded help topics because I found no place to put them.

For example, @gregmarr suggested a page on ADL.

Reply:
Right. I like the idea of https://documentation.divio.com but did not find the time to reorganize the documentation. Right now, I think just creating one page per topic is fine. I think it helps everybody:

- I often found bugs when I was forced to explain a concept in detail. Or I forgot an important requirement while writing a small example.
- I can reuse the docs very soon in issues or discussions, so it's an investment at least for me.
- A lot of questions never occur, because people can help themselves.
- mkdocs is providing a good-enough search, so even before cleaning up nothing is really "hidden".